src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -114,10 +131,7 @@ export function QueryPageContainerComponent(): JSX.Element {
         );
     }, [filter, table]);
 
-    let returnValue = <p> Loading data ... </p>;
-    if (_.isEmpty(data.ZNData) === false) {
-        returnValue = <QueryPageComponent />;
-    }
-
-    return returnValue;
+    return (
+        <ChooseComponent status={status} dataIsSet={!_.isEmpty(data.ZNData)} />

```

Comment:
Reject: the status already expresses if data was set, so dataIsSet is obsolete. You also check in this component for status code 200 and in ChooseComponent again. So you should decide which component is responsible for checking the status code and do it only once.
So in this state of the code (first commit) i would check the status here and remove the status property from ChooseComponent. The ChooseComponent does not need to know the status codes and which http request failed. It only needs to know if the app is loading, loading failed, or loading succeded.

Reply:
1. No it's not obsolete. React always render twice. If I do not check if the Data is already set, I get the error “Rendered more hooks than during the previous render” in the QueryPage.

Reply:
Ok, a similar error does exist in angular. It is check if your code is correctly designed. And if you get that message it is not. I would suggest to read some documentation about the message and try to understand why this error happens and where in your code is the problem. I'm not into this, but i would guess that you define the data hook only if the connection is a success, so your hooks change during render cycles. The solution might be to put the data hook in a child component that renders only if the connection succeeded. So in general the hooks should stick to the components, if the component is ready the hook should be too.

Reply:
Will be a future ToDo.